cli/internal/cloudcmd/upgrade.go

```diff
@@ -131,32 +142,41 @@ func (u *Upgrader) UpgradeNodeVersion(ctx context.Context, conf *config.Config)
 	if err := u.updateMeasurements(ctx, conf.GetMeasurements()); err != nil {
 		return fmt.Errorf("updating measurements: %w", err)
 	}
-	updatedNodeVersion, err := u.applyUpgrade(ctx, &components, nodeVersion)
+
+	updatedNodeVersion, err := u.applyNodeVersion(ctx, nodeVersion)
 	if err != nil {
 		return fmt.Errorf("applying upgrade: %w", err)
 	}
 	if updatedNodeVersion.Spec.ImageReference != imageReference ||
 		updatedNodeVersion.Spec.ImageVersion != imageVersion.Version ||
-		updatedNodeVersion.Spec.KubernetesComponentsReference != components.ObjectMeta.Name ||
+		updatedNodeVersion.Spec.KubernetesComponentsReference != nodeVersion.Spec.KubernetesComponentsReference ||
```

Comment:
I'd like to have those checks as separate if cases, which say which of those checks is false, and also both values. 

Use case: I forgot to set the correct config for `constellation upgrade apply` and it selected the wrong one by default which had a mismatch on the image. I know this should not happen for an end user, since they should have dedicated folders for each Constellation/config, but nevertheless, it is more verbose in such cases.